flocker/control/_model.py

```diff
@@ -410,34 +420,41 @@ class NodeState(PRecord):
         node that are currently shutting down or stopped.
     :ivar used_ports: A ``PSet`` of ``int``\ s giving the TCP port numbers
         in use (by anything) on this node.
-    :ivar PSet manifestations: All ``Manifestation`` instances that
-        are present on the node.
-    :ivar PMap paths: The filesystem paths of the manifestations on this
-        node. Maps ``dataset_id`` to a ``FilePath``.
     """
     hostname = field(type=unicode, factory=unicode, mandatory=True)
     used_ports = pset_field(int)
     running = pset_field(Application)
     not_running = pset_field(Application)
+
+    def update_deployment(self, deployment):
+        # Find the matching Node, replace its applications with
+        # combination of self.running and self.not_running
+        pass
+
+
+@implementer(IClusterStateUpdate)
+class NodeManifestations(PRecord):
+    """
+    The current state of a node's manifestations.
+
+    :ivar PSet manifestations: All ``Manifestation`` instances that
+        are present on the node.
+    :ivar PMap paths: The filesystem paths of the manifestations on this
+        node. Maps ``dataset_id`` to a ``FilePath``.
+    """
     manifestations = pset_field(Manifestation)
     paths = field(type=_PathMap, initial=_PathMap(), factory=_PathMap.create,
                   mandatory=True)
 
-    def to_node(self):
-        """
-        Convert into a ``Node`` instance.
-
-        :return Node: Equivalent ``Node`` object.
-        """
-        return Node(hostname=self.hostname,
-                    manifestations={m.dataset_id: m
-                                    for m in self.manifestations},
-                    applications=self.running | self.not_running)
+    def update_deployment(self, deployment):
+        # Find the matching Node, replace its manifestations with
+        # self.manifestations
+        pass
```

Comment:
This doesn't seem to agree with how `NodeManifestations` is being handled in `update_cluster_state`.  Maybe you meant to get rid of the `isinstance` check in that method (that seems like it would be kind of nice)?